ggircs-app/app/tests/unit/components/CarbonTaxActManagement/CarbonTaxActFuelType.test.tsx

```diff
@@ -0,0 +1,197 @@
+  replace: routerReplace,
+} as any);
+
+const loadedQuery = graphql`
+  query CarbonTaxActFuelTypeQuery($carbonTaxActFuelTypeId: ID!) {
+    query {
+      ...CarbonTaxActFuelType_query
+        @arguments(carbonTaxActFuelTypeId: $carbonTaxActFuelTypeId)
+    }
+  }
+`;
+
+let environment;
+const TestRenderer = () => {
+  const data = useLazyLoadQuery<CarbonTaxActFuelTypeQuery>(loadedQuery, {
+    carbonTaxActFuelTypeId: "test-fuel-id",
+  });
+  return <CarbonTaxActFuelType query={data.query} pageQuery={loadedQuery} />;
+};
+const renderCarbonTaxActFuelTypeComponent = () => {
+  return render(
+    <RelayEnvironmentProvider environment={environment}>
+      <TestRenderer />
+    </RelayEnvironmentProvider>
+  );
+};
+
+const getMockQueryPayload = () => ({
+  Query() {
+    return {
+      carbonTaxActFuelType: {
+        id: "test-fuel-id",
+        rowId: 1,
+        carbonTaxFuelType: "test-fuel",
+        ctaRateUnits: "test-units",
+        fuelChargesByCarbonTaxActFuelTypeId: {
+          __id: "connection-id",
+          edges: [
+            {
+              node: {
+                id: "test-charge-id",
+                startDate: "2020-01-01",
+                endDate: "2020-01-31",
+                fuelCharge: 0.5,
+                metadata: "test-metadata",
+              },
+            },
+          ],
+        },
+      },
+      allCarbonTaxActFuelTypes: {
+        allCarbonTaxActFuelTypes: {
+          edges: [
+            {
+              node: {
+                id: "test-fuel-id",
+                carbonTaxFuelType: "test-fuel",
+              },
+            },
+          ],
+        },
+      },
+    };
+  },
+});
+
+describe("The CarbonTaxActFuelType component", () => {
+  beforeAll(() => {
+    Object.defineProperty(window, "matchMedia", {
+      writable: true,
+      value: jest.fn().mockImplementation((query) => ({
+        matches: false,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(), // Deprecated
+        removeListener: jest.fn(), // Deprecated
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
+  });
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+
+    environment = createMockEnvironment();
+
+    environment.mock.queueOperationResolver((operation) =>
+      MockPayloadGenerator.generate(operation, getMockQueryPayload())
+    );
+
+    environment.mock.queuePendingOperation(compiledCarbonTaxActFuelTypeQuery, {
+      carbonTaxActFuelTypeId: "test-fuel-id",
+    });
+  });
+
+  it("Renders the component", () => {
+    renderCarbonTaxActFuelTypeComponent();
+
+    expect(screen.getByText("Select a carbon taxed fuel:")).toBeInTheDocument();
+  });
+
+  it("Calls the update mutation when editing a fuel charge", () => {
+    const mutationSpy = jest.fn();
+    jest
+      .spyOn(require("react-relay"), "useMutation")
+      .mockImplementation(() => [mutationSpy, jest.fn()]);
```

Comment:
for a tech debt card: assert the mutation was called on the `RelayMockEnvironment` instead

Reply:
taken care of when updating the icons -> buttons change